kernel/src/hil/analog_comparator.rs

```diff
@@ -0,0 +1,44 @@
+//! Interface for direct control of the analog comparators.
+
+// Author: Danilo Verhaert <verhaert@cs.stanford.edu>
+// Last modified 6/26/2018
+
+use returncode::ReturnCode;
+
+pub trait AnalogComparator {
+    /// Do a single comparison of two inputs, depending on the AC chosen. Output
+    /// will be True (1) when one is higher than the other, and False (0)
+    /// otherwise.  Specifically, the output is True when Vp > Vn (Vin positive
+    /// > Vin negative), and False if Vp < Vn.
+    fn comparison(&self, usize) -> bool;
```

Comment:
What is the `usize` argument to all of these functions? Presumably the analog comparator channel? I think it would be better to do something like what the ADC HIL does:

https://github.com/tock/tock/blob/f05997c2fdf1c2761a63efc5c612bef968ad6d2e/kernel/src/hil/adc.rs#L7-L10

where the type is specified by the specific implementation.